webapi/Controllers/ChatController.cs

```diff
@@ -145,6 +156,13 @@ public ChatController(ILogger<ChatController> logger, ITelemetryService telemetr
                 return this.BadRequest(string.Concat(aiException.Message, " - Detail: " + aiException.Detail));
             }
 
+            if (result.LastException is OperationCanceledException || result.LastException?.InnerException is OperationCanceledException)
+            {
+                // Log the timeout and return a 408 response
+                this._logger.LogError("The chat operation timed out.");
+                return this.StatusCode(StatusCodes.Status408RequestTimeout);
```

Comment:
You probably want 504 here, not 408.  408 means the client is the one giving the request too slowly and the server has had enough of your hogging a port.

504 means the server couldn't complete the request in time because one of its dependencies (e.g. AOAI) is taking too much time to return.

Reply:
Makes total sense. Changed! Thanks for calling this out!